main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -1087,10 +1087,34 @@ object Simplifier extends Phase[TypedAst.Root, SimplifiedAst.Root] {
               SimplifiedAst.Expression.Let(name, SimplifiedAst.Expression.Index(SimplifiedAst.Expression.Var(v, tpe, loc), idx, pat.tpe, loc), exp, succ.tpe, loc)
           }
 
+        /**
+          * Matching an array may succeed or fail
+          *
+          * We generate an if clause checking array length for each pattern, and then for each
+          * array element we generate a fresh variable and let-binding, which we then recurse over
```

Comment:
Explain what the let-binding binds, i.e. we generate a let-binding $x_i$ for each element in the array and load it with arrayload